pkg/jaeger/query/query.go

```diff
@@ -41,47 +45,85 @@ func (p *Query) SpanWriter() spanstore.Writer {
 }
 
 func (p *Query) GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error) {
+	metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": ""}).Inc()
+	start := time.Now()
 	res, err := getTrace(ctx, p.conn, traceID)
-	if err == nil {
-		traceRequestsExec.Add(1)
+	if err != nil {
+		metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": "500"}).Inc()
+		return nil, logError(err)
 	}
-	return res, logError(err)
+	traceRequestsExec.Add(1)
+	metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": "200"}).Inc()
+	metrics.RequestsDuration.With(prometheus.Labels{"type": "trace", "handler": "get_trace", "code": "200"}).Observe(time.Since(start).Seconds())
+	return res, nil
 }
 
 func (p *Query) GetServices(ctx context.Context) ([]string, error) {
+	metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_services", "code": ""}).Inc()
+	start := time.Now()
 	res, err := getServices(ctx, p.conn)
-	return res, logError(err)
+	if err != nil {
+		metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_services", "code": "500"}).Inc()
+		return nil, logError(err)
+	}
+	metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_services", "code": "200"}).Inc()
+	metrics.RequestsDuration.With(prometheus.Labels{"type": "trace", "handler": "get_services", "code": "200"}).Observe(time.Since(start).Seconds())
+	return res, nil
 }
 
 func (p *Query) GetOperations(ctx context.Context, query spanstore.OperationQueryParameters) ([]spanstore.Operation, error) {
+	metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_operations", "code": ""}).Inc()
+	start := time.Now()
 	res, err := getOperations(ctx, p.conn, query)
-	return res, logError(err)
+	if err != nil {
+		metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_operations", "code": "500"}).Inc()
+		return nil, logError(err)
+	}
+	metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_operations", "code": "200"}).Inc()
+	metrics.RequestsDuration.With(prometheus.Labels{"type": "trace", "handler": "get_operations", "code": "200"}).Observe(time.Since(start).Seconds())
+	return res, nil
```

Comment:
Same as 3 comments from `GetTrace()` instrumentation block

Reply:
Ideally, we would just use error message to set proper HTTP error code in both HTTP response and metric. However, I don't see this being possible with the current code. So what if we could reuse the pattern introduced in `InsertTraces()` function in the `trace` package?

Sth like:
```golang
func (p *Query) GetOperations(ctx context.Context, query spanstore.OperationQueryParameters) ([]spanstore.Operation, error) {
	code := "2xx"
	start := time.Now()
	defer func() {
		metrics.RequestsTotal.With(prometheus.Labels{"type": "trace", "handler": "get_operations", "code": code}).Inc()
		metrics.RequestsDuration.With(prometheus.Labels{"type": "trace", "handler": "get_operations", "code": code}).Observe(time.Since(start).Seconds())
	}()	
	res, err := getOperations(ctx, p.conn, query)
	if err != nil {
		code = "5xx"
		return nil, logError(err)
	}
	return res, nil
}
```